File: online-judge-solutions/Codeforces/1734A.py
#https://codeforces.com/problemset/problem/1734/A
#Lang		:	Python 3.8
#Time		:	46 ms
#Memory         :	0 KB
for _ in range(int(input())):
    n=int(input())
    a=list(map(int,input().split()))[:n]
    a.sort()
    r=1000000000
    r=[min(r,a[i+2]-a[i]) for i in range(n-2)]
    print(min(r))

File: online-judge-solutions/Codeforces/1734A.cpp
//https://codeforces.com/problemset/problem/1734/A
//Lang		:	GNU C++14
//Time		:	30 ms
//Memory	:	0 KB
#include<iostream>
#include<algorithm>
using namespace std;
int main(){
	int t,n;
	for(cin>>t;t--;){
		cin>>n;
		int a[n];
		for(int i=0;i<n;i++){cin>>a[i];}
		sort(a,a+n);
		int r=1000000000;
		for(int i=0;i<n-2;i++){
			r=min(r,a[i+2]-a[i]);
		}
		cout<<r<<endl;
	}
	return 0;
}
